src/utils/daos.c

```diff
@@ -786,6 +786,29 @@ obj_op_hdlr(struct cmd_args_s *ap)
 	return rc;
 }
 
+#define OCLASS_NAMES_LIST_SIZE 512
+
+static void
+print_oclass_names_list(FILE *stream)
+{
+	char *str;
+	size_t size = OCLASS_NAMES_LIST_SIZE, len;
+
+again:
+	str = malloc(size);
+	if (str == NULL)
+		return;
+	len = daos_oclass_names_list(size, str);
+	if (len <= 0)
+		return;
+	if (len < size)
+		fprintf(stream, "%s", str);
+	else {
+		size = len + 1;
```

Comment:
consider adding free(str) here to not leak the original allocation? I realize in this case the tool is probably just printing help and exiting in which case the leak becomes a non-issue.